transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollServerSocketChannel.java

```diff
@@ -53,17 +52,11 @@ public EpollServerSocketChannel(int fd) {
 
     EpollServerSocketChannel(LinuxSocket fd) {
         super(fd);
-        // As we create an EpollServerSocketChannel from a FileDescriptor we should try to obtain the remote and local
-        // address from it. This is needed as the FileDescriptor may be bound already.
-        local = fd.localAddress();
```

Comment:
After thinking about it a bit more it should be OK even if the user is doing this asynchronously because we would expect them to call the respective methods to initialize the socket:
- client: create the socket FD and then call connect
- server: create the socket FD and then call bind

So I think this is fine as is ... consider adding a comment to clarify when we accept a raw FD.